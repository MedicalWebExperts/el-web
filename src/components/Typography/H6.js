import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const H6 = ({ text }) => (
  <h1>{text}</h1>
);

H6.propTypes = propTypes;

export default H6;
