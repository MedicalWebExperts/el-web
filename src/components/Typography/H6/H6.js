import React from 'react';
import {
  string, bool, oneOf, shape,
} from 'prop-types';

import theme from '../../../styles';

const propTypes = {
  text: string.isRequired,
  secondary: bool,
  color: oneOf(['primary', 'secondary', 'default']),
  styles: shape({}),
};

const defaultProps = {
  color: 'default',
  secondary: false,
  styles: {},
};

const H6 = ({
  text, color, secondary, styles,
}) => (
  <h6 style={styles}>
    <style jsx>
      {`
        h6 {
          color: ${theme.colors[color]};
          font-family: ${secondary ? theme.font.secondary : theme.font.primary};
          font-size: ${theme.font.size}px;
          line-height: ${theme.font.size / 10};
          margin: 0;
        }
      `}
    </style>
    {text}
  </h6>
);

H6.propTypes = propTypes;
H6.defaultProps = defaultProps;

export default H6;
