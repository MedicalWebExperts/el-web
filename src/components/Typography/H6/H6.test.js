import React from 'react';
import renderer from 'react-test-renderer';

import H6 from './H6';

describe('Component H6 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<H6 text="Heading 6" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer.create(<H6 text="Heading 6" color="primary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer.create(<H6 text="Heading 6" secondary />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const myStyles = {
      marginTop: 30,
      lineHeight: 2,
    };
    const tree = renderer.create(<H6 text="Heading 1" styles={myStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
