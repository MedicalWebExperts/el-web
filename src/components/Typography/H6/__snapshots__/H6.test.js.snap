// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H6 Snapshot added optional style inline prop 1`] = `
<h6
  className="jsx-1380783411"
  style={
    Object {
      "lineHeight": 2,
      "marginTop": 30,
    }
  }
>
  Heading 1
</h6>
`;

exports[`Component H6 Snapshot primary color of the theme 1`] = `
<h6
  className="jsx-2927046201"
  style={Object {}}
>
  Heading 6
</h6>
`;

exports[`Component H6 Snapshot secondary font-family of the theme 1`] = `
<h6
  className="jsx-2405476878"
  style={Object {}}
>
  Heading 6
</h6>
`;

exports[`Component H6 Snapshot without arguments, take the default values 1`] = `
<h6
  className="jsx-1380783411"
  style={Object {}}
>
  Heading 6
</h6>
`;
